### PR TITLE
Series 2

### PR DESCRIPTION
**Pull request by cleargoal**

### CODE REVIEW

File: app/Http/Controllers/AdvertController.php [hunk 1]
```diff
@@ -41,7 +41,7 @@ public function store(CreateSubscriptionRequest $request): JsonResponse
      * @param Advert $advert
      * @return JsonResponse
      */
-    public function destroy(Request $request, Advert $advert): JsonResponse
+    public function destroy(Advert $advert, Request $request): JsonResponse
     {
         return response()->json($this->subscribeService->removeSubscription($request->user(), $advert));
```
Comment (ResetButton) [inline app/Http/Controllers/AdvertController.php:46]: Actually you can use auth()->user() wthout extra $request

Comment (R1KO) [reply app/Http/Controllers/AdvertController.php:46]: I think using global helpers is less flexible than explicit dependency injection

File: app/Http/Requests/CreateSubscriptionRequest.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+use Illuminate\Foundation\Http\FormRequest;
+
+class CreateSubscriptionRequest extends FormRequest
+{
+    public function authorize()
+    {
+        return true;
+    }
+
+    public function rules()
```
Comment (R1KO) [inline app/Http/Requests/CreateSubscriptionRequest.php:16]: `: array`

use type hinting wherever possible

File: app/Jobs/SendPriceNotificationJob.php [hunk 3]
```diff
@@ -28,10 +27,11 @@ public function __construct()
...
      * Execute the job.
      *
+     * @param MonitorService $monitorService
      * @return void
      */
-    public function handle(NotifyPriceService $notifyPriceService): void
+    public function handle(MonitorService $monitorService): void
     {
-        $notifyPriceService->process();
+        $monitorService->processPrices();
```
Comment (R1KO) [inline app/Jobs/SendPriceNotificationJob.php:35]: The name of this job does not match with method `processPrices`

File: app/Http/Requests/CreateSubscriptionRequest.php (hunk 2 above)
Comment (cleargoal) [reply app/Http/Requests/CreateSubscriptionRequest.php:16]: OK

**Pull request merged**